### PR TITLE
[INLONG-2920][DataProxy] Support keeping message order 

### LINKED ISSUE #2920: [Feature][Dataproxy] support keeping message order
### Description

support keeping message order

### Use case

support keeping message order

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by baomingyu**
support keeping message order 

Fixes #2920

### CODE REVIEW

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/utils/MessageUtils.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import org.apache.commons.lang.StringUtils;
+import org.apache.flume.Event;
+import org.apache.inlong.dataproxy.consts.AttributeConstants;
+import org.apache.inlong.dataproxy.source.MsgType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MessageUtils {
+
+    private static final Logger logger = LoggerFactory.getLogger(MessageUtils.class);
```
Comment (aloyszhang) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/utils/MessageUtils.java:30]: Seems `logger` is never used. Better to remove

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/channel/FailoverChannelSelector.java [hunk 2]
```diff
@@ -59,8 +63,16 @@
...
         } else if (event.getHeaders().containsKey(ConfigConstants.SLA_METRIC_DATA)) {
-
             retChannels.add(slaMetricChannels.get(0));
+        } else if (MessageUtils.isSyncSendForOrder(event.getHeaders()
+                .get(AttributeConstants.MESSAGE_SYNC_SEND))) {
+            String partitionKey = event.getHeaders().get(AttributeConstants.MESSAGE_PARTITION_KEY);
+            if (partitionKey == null) {
+                partitionKey = "";
+            }
+            int channelIndex = partitionKey.hashCode() % orderChannels.size();
```
Comment (aloyszhang) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/channel/FailoverChannelSelector.java:73]: A String value's hash code may be negative

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/channel/FailoverChannelSelector.java [hunk 3]
```diff
@@ -59,8 +63,16 @@
...
+            retChannels.add(orderChannels.get(channelIndex));
+            LOG.debug("use this!");
```
[the remark this answers is not in the document]
Comment (baomingyu) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/channel/FailoverChannelSelector.java]: done

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/utils/MessageUtils.java (hunk 1 above)
Comment (baomingyu) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/utils/MessageUtils.java]: Good!

**Review by dockerzhang — APPROVED** [review]

**Review by aloyszhang — APPROVED** [review]
LGTM

**Pull request merged**